syntax-markdown-commonmark12/src/main/java/org/xwiki/contrib/rendering/markdown/commonmark12/internal/renderer/MarkdownMacroRenderer.java

```diff
@@ -112,7 +114,11 @@ public String renderBlockMacro(String id, Map<String, String> parameters, String
 
     protected String renderBlockMacro(String id, Map<String, String> parameters, String content, boolean withNewLines)
     {
-        StringBuffer buffer = new StringBuffer();
+        if (id.equals(HTML_ID)) {
+            return content;
```

Comment:
What would happen here if you had an HTML macro containing something like `string // test // content`?

Reply:
I'm not sure I understand, are you referring to an HTML macro that would be put there explicitly by the user instead of after parsing HTML tags?

Reply:
Yes. As the result of the user's editing the HTML macro in the browser for example you could end up with content like `<strong>strange *test* content</strong>` that is displayed as <strong>strange \*test\* content</strong> before saving but without proper escaping you get <strong>strange *test* content</strong> after saving.

Reply:
This is now handled.